### PR TITLE
Controller

### PR DESCRIPTION
**Pull request by B-10515025**

### CODE REVIEW

**Comment by firewings2515** [comment]
Viewed, no problem for me

**Review by harrisonyei — APPROVED** [review]
LGTM

**Review by firewings2515 — APPROVED** [review]
Viewed, Look great to me

File: src/Components/Controller.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// MIT License
```
Comment (john0312) [inline src/Components/Controller.cpp:1]: Please run clang-format on this PR if you haven't done so.

File: src/Components/Controller.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#include "Controller.h"
+
+#include "Capture.h"
+#include "Decoder.h"
+#include "Encoder.h"
+#include "Render.h"
+#include "Router.h"
+
+void Controller::init(DisplayGLWidget* glWidget) {
+    render = new Render(glWidget);
```
Comment (john0312) [inline src/Components/Controller.cpp:35]: It is recommended to add comment on the data flow and its graph.

**Comment by ian287913** [comment]
LGTM!

File: src/Components/Controller.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+public:
+    // Default constructor
+    Controller() = default;
+    // Default destructor
+    ~Controller() = default;
+    // Set display widget pointer
+    void init(DisplayGLWidget*);
+    // Start streaming to target port 
+    void live(const int);
+    // Start receiving packet for source_ip:source_port and streaming ro output_port
```
Comment (john0312) [inline src/Components/Controller.h]: `to` ?

File: src/Components/Controller.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    void watch(const std::string&, const int, const int);
```
Comment (john0312) [inline src/Components/Controller.h]: Please add parameter naming to the definition, so people know which is source_port and which is output_port.

**Review by john0312 — APPROVED** [review]
Please address the comments before merging. Thank you.

**Review by ian287913 — APPROVED** [review]
LGTM

**Pull request merged**